pyhanko/sign/fields.py

```diff
@@ -1665,6 +1665,23 @@ def append_signature_field(
 
     if sig_field_spec.box is not None:
         llx, lly, urx, ury = sig_field_spec.box
+        matrix = None
+        if sig_field_spec.visible_sig_settings.rotate_with_page == False:
+            pagetree_obj = page_ref.get_object()
+            while '/Parent' in pagetree_obj and ('/Type' not in pagetree_obj or pagetree_obj['/Type'] != '/Page'):
+                pagetree_obj = pagetree_obj['/Parent']
+            if '/Rotate' in pagetree_obj:
+                match pagetree_obj['/Rotate']:
+                    case 0:
+                        matrix = (1.0, 0.0, 0.0, 1.0, 0.0, 0.0)
+                    case 90:
+                        matrix = (0.0, 1.0, -1.0, 0.0, 0.0, 0.0)
+                    case 180:
+                        matrix = (-1.0, 0.0, 0.0, -1.0, 0.0, 0.0)
+                    case 270:
+                        matrix = (0.0, -1.0, 1.0, 0.0, 0.0, 0.0)
+            else:
+                matrix = (1.0, 0.0, 0.0, 1.0, 0.0, 0.0)
```

Comment:
Can we skip the matrix entry altogether if it isn't needed? That is, when the rotation is 0 or unset.

Reply:
You are right, we can totally do that. I changed it